alacritty/src/window_context.rs

```diff
@@ -68,33 +75,111 @@ pub struct WindowContext {
 }
 
 impl WindowContext {
-    /// Create a new terminal window context.
-    pub fn new(
+    /// Create initial window context that dous bootstrapping the graphics Api we're going to use.
+    pub fn initial(
+        event_loop: &EventLoopWindowTarget<Event>,
+        proxy: EventLoopProxy<Event>,
         config: Rc<UiConfig>,
-        options: &WindowOptions,
-        window_event_loop: &EventLoopWindowTarget<Event>,
+        options: WindowOptions,
+        #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
+        wayland_event_queue: Option<&EventQueue>,
+    ) -> Result<Self, Box<dyn Error>> {
+        let raw_display_handle = event_loop.raw_display_handle();
+
+        let mut identity = config.window.identity.clone();
+        options.window_identity.override_identity_config(&mut identity);
+
+        // Windows has different order of GL platform initialization compared to any other platform,
+        // it requires the window first.
+        #[cfg(windows)]
+        let window = Window::new(event_loop, &config, &identity)?;
+
+        #[cfg(windows)]
+        let raw_window_handle = Some(window.raw_window_handle());
```

Comment:
```suggestion
        // Windows has different order of GL platform initialization compared to any other platform,
        // it requires the window first.
        #[cfg(windows)]
        let window = Window::new(event_loop, &config, &identity)?;
        #[cfg(windows)]
        let raw_window_handle = Some(window.raw_window_handle());
```

Probably better to keep the Windows stuff bundled together. Avoids accidentally having something written in between.